include/dsn/internal/message_parser.h

```diff
@@ -130,6 +130,10 @@ namespace dsn
         // return buffer count used, which must be no more than the return value of get_buffer_count_on_send().
         // may be invoked for mutiple times if the message is reused for resending.
         virtual int get_buffers_on_send(message_ex* msg, /*out*/ send_buf* buffers) = 0;
+
+    public:
+        static network_header_format get_header_type(const char* bytes); // buffer size >= sizeof(uint32_t)
+        static std::string get_debug_string(const char* bytes);
     };
 
     class message_parser_manager : public utils::singleton<message_parser_manager>
```

Comment:
we'd better init hdr_format value to "DSN_HDR_INVALID" in constructor of parser_factory_info.